webpack/assets/javascripts/bundle_novnc.js

```diff
@@ -0,0 +1,89 @@
+import RFB from '@novnc/novnc/core/rfb';
+
+let rfb;
+
+function sendCtrlAltDel() {
+  rfb.sendCtrlAltDel();
+  return false;
+}
+
+function showStatus(state, message) {
+  let level;
+  const status = $('#noVNC_status');
+  const ctrlAltDeleteButton = $('#ctrlAltDelButton');
+
+  switch (state) {
+    case 'failed':
+      level = 'danger';
+      break;
+    case 'fatal':
+      level = 'danger';
+      break;
+    case 'normal':
+      level = 'success';
+      break;
+    case 'disconnected':
+      level = 'default';
+      break;
+    default:
+      level = 'warning';
+      break;
+  }
+
+  ctrlAltDeleteButton.prop('disabled', state !== 'normal');
+
+  if (typeof message !== 'undefined') {
+    status.attr('class', `col-md-12 label label-${level}`);
+    status.html(message);
+  }
+}
+
+function securityFailed(e) {
+  let msg = '';
+  if ('reason' in e.detail) {
+    msg = `New connection has been rejected with reason: ${e.detail.reason}`;
+  } else {
+    msg = 'New connection has been rejected';
+  }
+  showStatus('fatal', msg);
+}
+
+function disconnectFinished() {
+  showStatus('failed', 'Disconnected');
+}
+
+function connectFinished() {
+  showStatus('normal', 'Connected');
+}
+
+$(document).on('ContentLoad', () => {
+  const vncScreen = $('#noVNC_screen');
+
+  if (vncScreen.length) {
+    $('#sendCtrlAltDelButton').on('click', sendCtrlAltDel);
+
+    const host = window.location.hostname;
+    const port = $('#vnc').attr('data-port');
+    const password = $('#vnc').attr('data-password');
+
+    let url;
+    if ($('#vnc').data('encrypt') === 'true') {
```

Comment:
How about:
```javascript
const protocol = $('#vnc').data('encrypt') === 'true' ? 'wss' : 'ws';
const host = window.location.hostname;
const port = $('#vnc').attr('data-port');
const url = `${protocol}://${host}:${port}`;
```

Reply:
Nice proposal, will change the code to this :-)